internal/stack/compose.go

```diff
@@ -111,7 +111,7 @@ func dockerComposeUp(options Options) error {
 
 	var args []string
 	if options.DaemonMode {
-		args = append(args, "-d")
+		args = append(args, "-d", "--wait", "--wait-timeout", fmt.Sprintf("%d", 600))
```

Comment:
@mrodm this is weird as the e2e-test in the CI have caught [one case that this happens](https://buildkite.com/elastic/elastic-package/builds/2584#018e3563-df58-4215-812a-49268dbabcfc/113-431). Can you help with an example that hopefully reproduces what you see?

Reply:
> this is weird as the e2e-test in the CI have caught [one case that this happens](https://buildkite.com/elastic/elastic-package/builds/2584#018e3563-df58-4215-812a-49268dbabcfc/113-431).

Steps in CI use detached mode (`-d`):
https://github.com/elastic/elastic-package/blob/caf7758c76f8e0aefcc9df159f8085ca86277299/scripts/test-check-packages.sh#L72

> Can you help with an example that hopefully reproduces what you see?

Sure ! Here I was referring to the case where that flag is not used:
```
elastic-package stack up -v
```

Not all runs of this command fail with this, so it needs to be repeated until it is hit that error.

After some retries running the above command, `elastic-agent-1` container could not start and `elastic-package` did not try to re-start the container as it would happen :

```
elastic-agent-1     | {"log.level":"info","@timestamp":"2024-03-26T11:51:50.649Z","log.origin":{"file.name":"cmd/enroll_cmd.go","file.line":505},"message":"1st enrollment attempt failed, retrying for 10m0s, every 1m0s enrolling to URL: https://fleet-server:8220/","ecs.version":"1.6.0"}
elastic-agent-1     | Error: fail to enroll: fail to execute request to fleet-server: EOF
elastic-agent-1     | For help, please see our troubleshooting guide at https://www.elastic.co/guide/en/fleet/8.12/fleet-troubleshooting.html
elastic-agent-1     | Error: enrollment failed: exit status 1
elastic-agent-1     | For help, please see our troubleshooting guide at https://www.elastic.co/guide/en/fleet/8.12/fleet-troubleshooting.html
elastic-agent-1 exited with code 1
```

And the status of the cluster:
```
 $ elastic-package stack status
Status of Elastic stack services:
╭──────────────────┬─────────┬───────────────────╮
│ SERVICE          │ VERSION │ STATUS            │
├──────────────────┼─────────┼───────────────────┤
│ elastic-agent    │ 8.12.2  │ exited (1)        │
│ elasticsearch    │ 8.12.2  │ running (healthy) │
│ fleet-server     │ 8.12.2  │ running (healthy) │
│ kibana           │ 8.12.2  │ running (healthy) │
│ package-registry │ latest  │ running (healthy) │
╰──────────────────┴─────────┴───────────────────╯
```

With the same options running with the latest published version, it does the retry:
```
elastic-agent-1              | {"log.level":"info","@timestamp":"2024-03-26T12:01:12.540Z","log.origin":{"file.name":"cmd/enroll_cmd.go","file.line":505},"message":"1st enrollment attempt failed, retrying for 10m0s, every 1m0s enrolling to URL: https://fleet-server:8220/","ecs.version":"1.6.0"}
elastic-agent-1              | Error: fail to enroll: fail to execute request to fleet-server: dial tcp 192.168.192.6:8220: connect: connection refused
elastic-agent-1              | For help, please see our troubleshooting guide at https://www.elastic.co/guide/en/fleet/8.12/fleet-troubleshooting.html
elastic-agent-1              | Error: enrollment failed: exit status 1
elastic-agent-1              | For help, please see our troubleshooting guide at https://www.elastic.co/guide/en/fleet/8.12/fleet-troubleshooting.html
elastic-agent-1 exited with code 1
dependency failed to start: container elastic-package-stack-elastic-agent-1 exited (1)
2024/03/26 13:01:12 DEBUG output command: /usr/bin/docker ps -a --filter label=com.docker.compose.project=elastic-package-stack --format {{.ID}}
2024/03/26 13:01:12 DEBUG output command: /usr/bin/docker inspect 0ebea2626c5e b5873a01c18c 94ec7181f20c 0449f8bf38ef a8c2d3eaf656 cc5633104400 1916c6b5f1dc 25baea050703 b1af9e72d4f2 2f268ae81ff3
Elastic Agent failed to start, trying again in 10s.
2024/03/26 13:01:22 DEBUG running command: /usr/bin/docker compose version --short
2024/03/26 13:01:22 DEBUG Determined Docker Compose version: 2.25.0
2024/03/26 13:01:22 DEBUG running command: /usr/bin/docker compose -f /home/mariorodriguez/.elastic-package/profiles/default/stack/snapshot.yml -p elastic-package-stack up
WARN[0000] /home/mariorodriguez/.elastic-package/profiles/default/stack/snapshot.yml: `version` is obsolete 
[+] Running 9/0
 ✔ Container elastic-package-stack-package-registry-1           Running                                                                           0.0s 
 ✔ Container elastic-package-stack-elasticsearch-1              Running                                                                           0.0s 
 ✔ Container elastic-package-stack-package-registry_is_ready-1  Created                                                                           0.0s 
 ✔ Container elastic-package-stack-elasticsearch_is_ready-1     Created                                                                           0.0s 
 ✔ Container elastic-package-stack-kibana-1                     Running                                                                           0.0s 
 ✔ Container elastic-package-stack-kibana_is_ready-1            Created                                                                           0.0s 
 ✔ Container elastic-package-stack-fleet-server-1               Running                                                                           0.0s 
 ✔ Container elastic-package-stack-fleet-server_is_ready-1      Created                                                                           0.0s 
 ✔ Container elastic-package-stack-elastic-agent-1              Created                                                                           0.0s 
Attaching to elastic-agent-1, elastic-agent_is_ready-1, elasticsearch-1, elasticsearch_is_ready-1, fleet-server-1, fleet-server_is_ready-1, kibana-1, kibana_is_ready-1, package-registry-1, package-registry_is_ready-1
```

Reply:
that's odd 🥲 hmmm I see, I think this is what happens; with the *_is_ready services in place we have dependencies on the actual services, so when elastic-agent fails the respective _is_ready service can't start because it's dependency failed completely and thus docker compose up return an error in the sense sorry I couldn't bring up all the services. However when we remove the *_is_ready services then this isn't triggering as nothing depends on elastic-agent and thus up considers that everything had been brought up and the user will deal with any errors visible in the logs... I don't have an immediate fix for that